src/lib.rs

```diff
@@ -17,87 +21,189 @@ impl Default for Syntax {
+    ///
+    /// Uses variable-length encoding. Signed int value. These more efficiently
+    /// encode negative numbers than regular int32s.
     Sint32,
+    /// Protobuf sint64
+    ///
+    /// # Remarks
+    ///
+    /// Uses variable-length encoding. Signed int value. These more efficiently
+    /// encode negative numbers than regular int32s.
     Sint64,
+    /// Protobuf bool
     Bool,
+    /// Protobuf enum (holds the enum name)
     Enum(String),
+    /// Protobuf fixed64
+    ///
+    /// # Remarks
+    ///
+    /// Always eight bytes. More efficient than uint64 if values are often greater than 2^56.
     Fixed64,
+    /// Protobuf sfixed64
+    ///
+    /// # Remarks
+    ///
+    /// Always eight bytes.
     Sfixed64,
+    /// Protobuf double
     Double,
+    /// Protobuf string
+    ///
+    /// # Remarks
+    ///
+    /// A string must always contain UTF-8 encoded or 7-bit ASCII text.
     String_,
+    /// Protobuf bytes
+    ///
+    /// # Remarks
+    ///
+    /// May contain any arbitrary sequence of bytes.
     Bytes,
+    /// Protobut message (holds the message name)
     Message(String),
+    /// Protobut fixed32
+    ///
+    /// # Remarks
+    ///
+    /// Always four bytes. More efficient than uint32 if values are often greater than 2^28.
     Fixed32,
+    /// Protobut sfixed32
+    ///
+    /// # Remarks
+    ///
+    /// Always four bytes.
     Sfixed32,
+    /// Protobut float
     Float,
+    /// Protobut map
     Map(Box<(FieldType, FieldType)>),
 }
 
+/// A Protobuf Field
 #[derive(Debug, Clone)]
 pub struct Field {
+    /// Field name
     pub name: String,
+    /// Field `Frequency`
     pub frequency: Frequency,
+    /// Field type
     pub typ: FieldType,
+    /// Tag number
     pub number: i32,
+    /// Default value for the field
     pub default: Option<String>,
+    /// Packed property for repeated fields
     pub packed: Option<bool>,
-    pub boxed: bool,
+    /// Is the field deprecated
     pub deprecated: bool,
 }
 
+/// A protobuf message
 #[derive(Debug, Clone, Default)]
 pub struct Message {
+    /// Message name
     pub name: String,
+    /// Message `Field`s
     pub fields: Vec<Field>,
+    /// Message `OneOf`s
     pub oneofs: Vec<OneOf>,
+    /// Message reserved numbers
     pub reserved_nums: Option<Vec<i32>>,
+    /// Message reserved names
     pub reserved_names: Option<Vec<String>>,
-    pub imported: bool,
-    pub package: String,        // package from imports + nested items
-    pub messages: Vec<Message>, // nested messages
-    pub enums: Vec<Enumerator>, // nested enums
-    pub module: String,         // 'package' corresponding to actual generated Rust module
+    /// Nested messages
+    pub messages: Vec<Message>,
+    /// Nested enums
+    pub enums: Vec<Enumerator>,
 }
 
+/// A protobuf enumerator field
+#[derive(Debug, Clone)]
+pub struct EnumVariant {
+    /// enum variant name
+    pub name: String,
+    /// enum variant number
+    pub number: i32,
+}
```

Comment:
Enum value can be deprecated, worth adding a TODO at least.

Reply:
Are you sure? In the [doc](https://developers.google.com/protocol-buffers/docs/proto3), the `deprecated` option is only for "field option" (not enum values).

Reply:
OK, I see this option presents in `descriptor.proto`, but probably they didn't pull it to parser. So it's fine without the option.